utils/gridlink_impl.c.src

```diff
@@ -369,7 +357,9 @@ cellarray_DOUBLE * gridlink_DOUBLE(const int64_t NPART,
                 SGLIB_ARRAY_ELEMENTS_EXCHANGER(DOUBLE, first->x, i, j); \
                 SGLIB_ARRAY_ELEMENTS_EXCHANGER(DOUBLE, first->y, i, j); \
                 SGLIB_ARRAY_ELEMENTS_EXCHANGER(DOUBLE, first->z, i, j); \
-                SGLIB_ARRAY_ELEMENTS_EXCHANGER(int64_t, first->original_index, i, j); \
+                if(!options->copy_particles) { \
```

Comment:
Not sure I follow this change - seems to depart from the previous code. Will you please add a clarifying note why the `original_index` is only sorted when `copy_particles == 0`.